ribs.py

```diff
@@ -167,6 +242,28 @@ def main():
             y = math.sin(a) * r + SCREEN_WIDTH / 2
             draw_transformed(teapot, (x, y), (0.5, 2.0), a * 180 / math.pi)
 
+        if pg.mouse.get_pressed()[0]:
+            vel_a = ((rect_a.centerx - pg.mouse.get_pos()[0]) / -DELTA,
+                     (rect_a.centery - pg.mouse.get_pos()[1]) / -DELTA)
+            rect_a.center = pg.mouse.get_pos()
+        rect_a, rect_b, vel_a, vel_b, hit = solve_rect_overlap(rect_a, rect_b, vel_a, vel_b, bounce=1.0)
+
+        print(vel_a, vel_b)
+        vel_b = damping(vel_b, 0.1)
+        rect_b.x += int(vel_b[0] * DELTA)
+        rect_b.y += int(vel_b[1] * DELTA)
+
+        if not pg.mouse.get_pressed()[0]:
+            rect_a.x += int(vel_a[0] * DELTA)
+            rect_a.y += int(vel_a[1] * DELTA)
+
+        window = pg.display.get_surface()
+        pg.draw.rect(window, pg.Color(255, 200, 200), rect_a)
+        if hit:
+            pg.draw.rect(window, pg.Color(200, 255, 255), rect_b)
+        else:
+            pg.draw.rect(window, pg.Color(200, 0, 255), rect_b)
+
```

Comment:
This code could use some comments saying what each block does in my opinion. It's quite hard to tell without good understanding of the code

Reply:
In my head, this is "boilerplate testing code" I am expecting it to be torn out and replaced with a simple game. So I'm not too worried about commenting it. Unless you think it's good example code, but we might need a place to put it first. ;)